doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,385 @@
+
+```js
+{
+  "profiles": [
+    {
+      "updates": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
+      "fontSize": 16,
+      "fontWeight": "thin"
+    },
+    {
+      "guid": "{a821ae62-9d4a-3e34-b989-0a998ec283e6}",
+      "name": "Cool Profile",
+      "commandline": "powershell.exe",
+      "antialiasingMode": "aliased",
+      "fontWeight": "bold",
+      "scrollbarState": "hidden"
+    }
+  ],
+  "schemes": [
+    {
+        "name": "Postmodern Tango Light",
+
+        "cursorColor": "#FFFFFF",
+        "selectionBackground": "#FFFFFF",
+
+        "background": '#61D6D6',
+        "foreground": '#E74856',
+
+        "black" : "#0C0C0C",
+        "blue" : "#0037DA",
+        "cyan" : "#3A96DD",
+        "green" : "#13A10E",
+        "purple" : "#881798",
+        "red" : "#C50F1F",
+        "white" : "#CCCCCC",
+        "yellow" : "#C19C00",
+        "brightBlack" : "#767676",
+        "brightBlue" : "#3B78FF",
+        "brightCyan" : "#61D6D6",
+        "brightGreen" : "#16C60C",
+        "brightPurple" : "#B4009E",
+        "brightRed" : "#E74856",
+        "brightWhite" : "#F2F2F2",
+        "brightYellow" : "#F9F1A5"
+    }
+  ]
+}
+```
+
+### Creation and location(s) of the json files
+
+In this section, we cover where an app that wants to use this proto-extension functionality should put the json
+files. Once we implement this feature, we will need to provide documentation on this for app developers. 
+
+#### For apps installed through Microsoft store (or similar)
+
+For apps that are installed through something like the Microsoft Store, they will need to declare themselves to
+be an app extension or write a separate app extension. In the app extension, they will need to create a public
+folder, and create a subdirectory within that folder called `Fragments`. The json files should be inserted into the
+`Fragments` subdirectory. The specifics of how they can declare themselves as an app extension are provided in the
+[Microsoft Docs](https://docs.microsoft.com/en-us/windows/uwp/launch-resume/how-to-create-an-extension), and I will
+replicate the necessary section here. 
+
+In the appxmanifest file of the package:
+
+```
+<Package
+  ...
+  xmlns:uap3="http://schemas.microsoft.com/appx/manifest/uap/windows10/3"
+  IgnorableNamespaces="uap uap3 mp">
+  ...
+    <Applications>
+      <Application Id="App" ... >
+        ...
+        <Extensions>
+          ...
+          <uap3:Extension Category="windows.appExtension">
+            <uap3:AppExtension Name="com.Microsoft.Terminal"
+                               Id="<id>"
+                               PublicFolder="Public">
+            </uap3:AppExtension>
+          </uap3:Extension>
+        </Extensions>
+      </Application>
+    </Applications>
+    ...
+</Package>
+```
+
+Note that the name field **must** be `com.Microsoft.Terminal` for us to detect this extension. The `Id` field

```

Comment:
Please also add a subcategory like `com.microsoft.terminal.settingsext` because I intend to use `com.microsoft.terminal.host` for the 3rd party terminal hosts and `com.microsoft.console.host` for 3rd party console host registrations.

Reply:
Urgh, I used "com.microsoft.windows.terminal.host" in mine, so maybe "com.microsoft.windows.terminal.settingsext" or something. We should just be consistent with each other.

Reply:
> setting sext

Maybe just `settings`?